components/ArticleReply/index.js

```diff
@@ -0,0 +1,242 @@
+
+const useStyles = makeStyles(theme => ({
+  root: {
+    '&:not(:first)': {
+      borderTop: `1px dotted ${theme.palette.secondary[100]}`,
+    },
+  },
+  replyType: {
+    color: ({ replyType }) => {
+      switch (replyType) {
+        case 'OPINIONATED':
+          return '#2079F0';
+        case 'NOT_RUMOR':
+          return '#00B172';
+        case 'RUMOR':
+          return '#FB5959';
+        default:
+          return theme.palette.common.black;
+      }
+    },
+  },
+  content: {
+    padding: '17px 0',
+    borderBottom: `1px dotted ${theme.palette.secondary[100]}`,
+  },
+}));
+
+const ArticleReplyData = gql`
+  fragment ArticleReplyData on ArticleReply {
+    # articleId and replyId are required to identify ArticleReply instances
+    articleId
+    replyId
+    canUpdateStatus
+    createdAt
+    reply {
+      id
+      type
+      text
+      reference
+      user {
+        id
+        name
+        level
+        avatarUrl
+      }
+      hyperlinks {
+        ...HyperlinkData
+      }
+      ...ReplyInfo
+    }
+    user {
+      id
+      name
+      level
+      avatarUrl
+    }
+    ...ArticleReplyFeedbackData
+  }
+  ${Hyperlinks.fragments.HyperlinkData}
+  ${ReplyFeedback.fragments.ArticleReplyFeedbackData}
+  ${ReplyInfo.fragments.replyInfo}
+`;
+
+const ArticleReplyForUser = gql`
+  fragment ArticleReplyForUser on ArticleReply {
+    # articleId and replyId are required to identify ArticleReply instances
+    articleId
+    replyId
+    canUpdateStatus
+    ...ArticleReplyFeedbackForUser
+  }
+  ${ReplyFeedback.fragments.ArticleReplyFeedbackForUser}
+`;
+
+const ArticleReply = React.memo(
+  ({
+    articleReply = {},
+    disabled = false,
+    onAction = () => {},
+    actionText = '',
+    showActionOnlyWhenCanUpdate = true, // If false, show action button for everyone
+    showFeedback = true,
+  }) => {
+    const {
+      articleId,
+      positiveFeedbackCount,
+      negativeFeedbackCount,
+      feedbacks,
+      reply,
+      replyId,
+      ownVote,
+      user: articleReplyAuthor,
+    } = articleReply;
+
+    const { type: replyType } = reply;
+
+    const replyAuthor = reply.user;
+    const user = replyAuthor || articleReplyAuthor;
+
+    const classes = useStyles({ replyType });
+
+    const renderFooter = useCallback(() => {
+      const copyText =
+        typeof window !== 'undefined'
+          ? `${TYPE_NAME[reply.type]} \n【${t`Reason`}】${(
+              reply.text || ''
+            ).trim()}\n↓${t`Details`}↓\n${
+              window.location.href
+            }\n↓${t`Reference`}↓\n${reply.reference}`
+          : '';
+
+      return (
+        <Box component="footer" display="flex" py={2}>
+          {showFeedback && (
+            <ReplyFeedback
+              articleId={articleId}
+              replyId={replyId}
+              positiveFeedbackCount={positiveFeedbackCount}
+              negativeFeedbackCount={negativeFeedbackCount}
+              feedbacks={feedbacks}
```

Comment:
I found a weird case that my feedback is visible in API but not in UI.

![螢幕快照 2020-05-17 上午12 48 24](https://user-images.githubusercontent.com/108608/82125422-6fa47380-97d8-11ea-8f9c-c034e02abcd6.png)

Reply:
I think it's because that the logic here is to display feebacks with user present:
```
{feedbacks
            .filter(({ vote, user }) => vote === 'UPVOTE' && user)
            .map(feedback => (
              <Feedback key={feedback.id} {...feedback} />
            ))}
```

Reply:
Got it, make sense.
I am going to [revamp user handling](https://g0v.hackmd.io/ZcoUOX_-RQSkJyl5xz4_Zg), giving users from `localhost` a dedicated browser app ID on staging site, and show browser app users to all apps. Hopefully this will solve these weird issue during development.